esp-box/idf_component.yml

```diff
@@ -1,4 +1,4 @@
-version: "2.3.0"
+version: "2.4.0"
```

Comment:
It isn't breaking change. Do we need change this version. I think, that enough for this change is version 2.3.1. (same for all BSPs edited in this PR)

Reply:
Thank you for the review @espzav 

Breaking changes should be indicated by major version bump (->3.0.0).

But I agree, let's bump to 2.3.1